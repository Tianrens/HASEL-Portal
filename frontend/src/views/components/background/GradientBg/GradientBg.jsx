import React from 'react';

import { heroPattern, container } from './GradientBg.module.scss';

function GradientBg({ children }) {
    return (
        <div className={heroPattern}>
            <div className={container}>{children}</div>
        </div>
    );
}

export default GradientBg;
