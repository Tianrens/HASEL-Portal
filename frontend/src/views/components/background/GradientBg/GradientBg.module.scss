.heroPattern {
    background-color: #ffffff;
    background-image: url("data:image/svg+xml,%3Csvg width='20' height='20' viewBox='0 0 20 20' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='%23dadada' fill-opacity='0.36' fill-rule='evenodd'%3E%3Ccircle cx='3' cy='3' r='3'/%3E%3Ccircle cx='13' cy='13' r='3'/%3E%3C/g%3E%3C/svg%3E");
    display: flex;
    flex: 1;
}

.container {
    background: linear-gradient(to bottom, #fff 0%, rgba(0, 0, 0, 0) 100%);
    display: flex;
    flex-direction: column;
    width: 100%;
}
