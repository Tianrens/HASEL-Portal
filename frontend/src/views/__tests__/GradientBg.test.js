import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider } from '@material-ui/core';
import GradientBg from '../components/background/GradientBg/GradientBg';

test('Gradient Background renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <GradientBg />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
