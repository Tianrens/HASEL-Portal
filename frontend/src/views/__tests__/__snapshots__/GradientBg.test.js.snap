// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gradient Background renders properly 1`] = `
<div
  className="heroPattern"
>
  <div
    className="container"
  />
</div>
`;
